components/context_storage/IContextStorage.py

```diff
@@ -0,0 +1,46 @@
+from abc import ABC, abstractmethod
```

Comment:
This "interface" file seems like it shouldn't have to know anything about the various storage options. Can't we change this such that pinecode/weaviate implement an agnostic interface and contain all of the vector store specifics in their respective files?

Reply:
Correct. Also, why IContextStorage vs just ContextStorage? I think the latter is more idiomatic as per pep8?

Reply:
Anyway, happy to help, I think this is a step in the right direction, just need to get it in and we can iterate. @atroyn

Reply:
We could in principle take the constructor method out and break this up further, but I'm not sure the modularity is warranted yet.